### DISCUSSION TITLE
Problem with protein ligand gmx_mmpbsa calculation

### DISCUSSION THREAD

**Discussion by drsandhyaks** [Valdes-Tresanco-MS/gmx_MMPBSA#216] [Q&A] [answered]
Dear Valdes,
 I conducted MD in charm 27 for protein and for ligand forcefield is from PRODRUG server
this command i used. in gromacs 2021 
gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ci index.ndx -cg 22 23 -ct md_0_1.xtc -cp topol.top
I got the error which is given
[INFO   ] Starting
[INFO   ] Command-line
  gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ci index.ndx -cg 22 23 -ct md_0_1.xtc -cp topol.top

[WARNING] protein_forcefield and ligand_forcefield variables are deprecate since version 1.4.1 and will be remove in the next version. Please,                                               use forcefield instead.
[WARNING] entropy_seg variable is deprecate since version 1.4.2 and will be remove in v1.5.0. Please, use ie_segment instead.
[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /opt/anaconda3/bin/cpptraj
[INFO   ] tleap found! Using /opt/anaconda3/bin/tleap
[INFO   ] parmchk2 found! Using /opt/anaconda3/bin/parmchk2
[INFO   ] sander found! Using /opt/anaconda3/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/local/gromacs/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER Topologies from GROMACS files...
[INFO   ] Checking gmxMMPBSA data folder exists in Amber data...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group 22_23 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[INFO   ] No receptor structure file was defined. Using ST approach...
[INFO   ] Using receptor structure from complex to generate AMBER topology
[INFO   ] Normal Complex: Saving group 22 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_REC.pdb
[INFO   ] No ligand structure file was defined. Using ST approach...
[INFO   ] Using ligand structure from complex to generate AMBER topology
[INFO   ] Normal ligand: Saving group 23 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_LIG.pdb
[INFO   ] Building Normal Complex Amber Topology...
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:500: GromacsWarning: bond funct != 1; unknown functional
  warnings.warn('bond funct != 1; unknown functional',
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py:548: GromacsWarning: angles funct != 1 or 5; unknown functional
  warnings.warn('angles funct != 1 or 5; unknown '
  File "/opt/anaconda3/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/app.py", line 95, in gmxmmpbsa
    app.make_prmtops()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/main.py", line 576, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/make_top.py", line 117, in buildTopology
    tops = self.gmxtop2prmtop()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/make_top.py", line 318, in gmxtop2prmtop
    com_top = self.cleantop(self.FILES.complex_top, self.indexes[0] + self.indexes[1])
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/make_top.py", line 598, in cleantop
    rtemp_top = parmed.gromacs.GromacsTopologyFile(ttp_file.as_posix())
  File "/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py", line 251, in __init__
    self.read(fname, defines, parametrize)
  File "/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py", line 460, in read
    self.parametrize()
  File "/opt/anaconda3/lib/python3.8/site-packages/parmed/gromacs/gromacstop.py", line 1013, in parametrize
    raise ParameterError('Not all bond parameters found')
ParameterError: Not all bond parameters found
Exiting. All files have been retained.

I think it is the problem with topology of ligand.
How to rectifiy this??

**Comment by marioernestovaldes** [comment, marked as answer]
The error comes from using GROMOS force field parameters which are not supported in gmx_MMPBSA at the moment... isn't PRODRUG server used to generate GROMOS compatible parameters for small ligands? if you are using charmm27 for the protein, the ligand should be parameterised with CGenFF

**Comment by drsandhyaks** [comment]
Thanks.
I tried with charm27 and swissparm for ligand..
But i got this error.
[INFO   ] Starting
[INFO   ] Command-line
  gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ci index.ndx -cg 1 13 -ct md_0_1.xtc -lm LIG.mol2

[WARNING] protein_forcefield and ligand_forcefield variables are deprecate since version 1.4.1 and will b                                                                                    e remove in the next version. Please, use forcefield instead.
[WARNING] entropy_seg variable is deprecate since version 1.4.2 and will be remove in v1.5.0. Please, use                                                                                     ie_segment instead.
[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /opt/anaconda3/bin/cpptraj
[INFO   ] tleap found! Using /opt/anaconda3/bin/tleap
[INFO   ] parmchk2 found! Using /opt/anaconda3/bin/parmchk2
[INFO   ] mmpbsa_py_energy found! Using /opt/anaconda3/bin/mmpbsa_py_energy
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/local/gromacs/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER Topologies from GROMACS files...
[INFO   ] Checking gmxMMPBSA data folder exists in Amber data...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group 1_13 in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[INFO   ] Generating ligand parameters from LIG.mol2 file...
[ERROR  ] MMPBSA_Error /opt/anaconda3/bin/parmchk2 failed when querying LIG.mol2.
           Check the gmx_MMPBSA.log file to report the problem.
  File "/opt/anaconda3/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/app.py", line 95, in gmxmmpbsa
    app.make_prmtops()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/main.py", line 576, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/make_top.py", line 115, in buildTopology
    self.gmx2pdb()
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/make_top.py", line 181, in gmx2pdb
    GMXMMPBSA_ERROR('%s failed when querying %s' % (parmchk2, self.FILES.ligand_mol2))
  File "/opt/anaconda3/lib/python3.8/site-packages/GMXMMPBSA/exceptions.py", line 169, in __init__
    raise exc(msg + '. Check the gmx_MMPBSA.log file to report the problem.')
MMPBSA_Error: /opt/anaconda3/bin/parmchk2 failed when querying LIG.mol2. Check the gmx_MMPBSA.log file to                                                                                     report the problem.
Exiting. All files have been retained.
(base) dcbsandhya@kugpu1:~/O/1e3g_F$ vi gmx_MMPBSA.log
as published by the Free Software Foundation; either version 2.1
of the License, or (at your option) any later version.

GROMACS:      gmx editconf, version 2021
Executable:   /usr/local/gromacs/bin/gmx_mpi
Data prefix:  /usr/local/gromacs
Working dir:  /home/dcbsandhya/O/1e3g_F
Command line:
  gmx editconf -f md_0_1.tpr -o _GMXMMPBSA_COM.pdb -n _GMXMMPBSA_COM_index.ndx

Reading file md_0_1.tpr, VERSION 2021 (single precision)
Reading file md_0_1.tpr, VERSION 2021 (single precision)

Select a group for output:
Group     0 (         System) has 76236 elements
Group     1 (  GMXMMPBSA_REC) has  4108 elements
Group     2 (      Protein-H) has  2044 elements
Group     3 (        C-alpha) has   250 elements
Group     4 (       Backbone) has   750 elements
Group     5 (      MainChain) has   999 elements
Group     6 (   MainChain+Cb) has  1240 elements
Group     7 (    MainChain+H) has  1239 elements
Group     8 (      SideChain) has  2869 elements
Group     9 (    SideChain-H) has  1045 elements
Group    10 (    Prot-Masses) has  4108 elements
Group    11 (    non-Protein) has 72128 elements
Group    12 (          Other) has    30 elements
Group    13 (  GMXMMPBSA_LIG) has    30 elements
Group    14 (             CL) has     5 elements
Group    15 (          Water) has 72093 elements
Group    16 (            SOL) has 72093 elements
Group    17 (      non-Water) has  4143 elements
Group    18 (            Ion) has     5 elements
Group    19 (            LIG) has    30 elements
Group    20 (             CL) has     5 elements
Group    21 ( Water_and_ions) has 72098 elements
Group    22 (    Protein_LIG) has  4138 elements
Group    23 (GMXMMPBSA_REC_GMXMMPBSA_LIG) has  4138 elements
Select a group:
GROMACS reminds you: "Wicky-wicky Wa-wild West" (Will Smith)

Note that major changes are planned in future for editconf, to improve usability and utility.
Read 76236 atoms
Volume: 773.077 nm^3, corresponds to roughly 347800 electrons
Velocities found
Selected 23: 'GMXMMPBSA_REC_GMXMMPBSA_LIG'
Segmentation fault (core dumped)
~


given below is mmpbsa.in

Sample input file for GB calculation
This input file is meant to show only that gmx_MMPBSA works. Althought,
we tried to used the input files as recommended in the Amber manual,
some parameters have been changed to perform more expensive calculations
in a reasonable amount of time. Feel free to change the parameters
according to what is better for your system.

&general
sys_name="Prot-LIG",
startframe=49000,
endframe=50000,
interval=100,
/
&gb
igb=5, saltcon=0.150,
/

**Comment by marioernestovaldes** [comment]
if you are using charmm force field, you must define the topology file with the flag "-cp"... please, check this [tutorial](https://valdes-tresanco-ms.github.io/gmx_MMPBSA/examples/Protein_ligand_CHARMMff/)

cheers!

**Comment by drsandhyaks** [comment]
thank you so much now works..
🙏
